react-app/src/utils/leaguehelpers.js

```diff
@@ -33,29 +33,36 @@ export default {
 
 	getDate: (league, dateId) => {
 		//dateId: league.scheduledMatches.dateId
-		if(league == null || dateId === ''){
+		var getdate = {};
+
+		if(league === null || dateId === ''){
 			return null;
 		}
 
 		league.dates.forEach((date) => {
```

Comment:
instead of doing a forEach here you can do a league.dates.filter((date) => {});. Filter reduces the array into an object and if it's not set just return {}.